components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/RedisSessionDataStore.java

```diff
@@ -0,0 +1,277 @@
+     * mentioned in IDENTITY-5131
+     */
+    public void removeExpiredOperationData() {
+        // Empty method
+    }
+
+    @Override
+    /**
+     * {@inheritDoc}
+     */
+    public SessionContextDO getSessionContextData(String key, String type) {
+
+        Jedis jedis;
+        String redisKey = key + Constants.DIVIDER + type;
+        if (!enablePersist) {
+            return null;
+        }
+        boolean tempStore = getSessionStoreType(type);
+        if (tempStore) {
+            redisKey = redisKey + Constants.DIVIDER + Constants.TEMPSTORE;
+        }
+        try {
+            jedis = getJedisInstance().getResource();
+            if (jedis.exists(redisKey)) {
+                long nanoTime = Long.parseLong(jedis.hget(redisKey, Constants.NANO_TIME));
+                String objectKey = redisKey + Constants.DIVIDER + Constants.OBJECT;
+                Object blobObject = deserialize(jedis.get(objectKey.getBytes()));
+                jedis.close();
+                return new SessionContextDO(key, type, blobObject, nanoTime);
+            }
+            return null;
+        } catch (JedisException e) {
+            log.error("Error while retrieving session data", e);
+        }
+        return null;
+    }
+
+    @Override
+    /**
+     * {@inheritDoc}
+     */
+    public void persistSessionData(String key, String type, Object entry, long nanoTime, int tenantId) {
+
+        long validityPeriodNano = 0L;
+        String objectKey;
+        Long expireTime;
+        String redisKey;
+        boolean tempStore = getSessionStoreType(type);
+        redisKey = key + Constants.DIVIDER + type;
+        if (!enablePersist) {
+            return;
+        }
+        if (entry instanceof CacheEntry) {
+            validityPeriodNano = ((CacheEntry) entry).getValidityPeriod();
+        }
+        if (validityPeriodNano == 0L) {
+            validityPeriodNano = getCleanupTimeout(type, tenantId);
+        }
+        if (tempStore) {
+            redisKey = redisKey + Constants.DIVIDER + Constants.TEMPSTORE;
+        }
+        try {
+            Jedis jedis;
+            jedis = getJedisInstance().getResource();
+            jedis.hset(redisKey, Constants.NANO_TIME, String.valueOf(nanoTime));
+            jedis.hset(redisKey, Constants.EXPIRY_TIME, String.valueOf(nanoTime + validityPeriodNano));
+            jedis.hset(redisKey, Constants.TENANT_ID, String.valueOf(tenantId));
+            objectKey = redisKey + Constants.DIVIDER + Constants.OBJECT;
+            byte[] serializedSessionObject = serialize((Serializable) entry);
+            jedis.set(objectKey.getBytes(), serializedSessionObject);
+
+            expireTime = TimeUnit.SECONDS.convert(validityPeriodNano, TimeUnit.NANOSECONDS);
+            jedis.expire(redisKey, expireTime.intValue());
+            jedis.expire(objectKey, expireTime.intValue());
+            jedis.close();
```

Comment:
Let's extract this out to a private method.

Reply:
But this is used only once, so I thought there is no much impact by extracting out this logic there.